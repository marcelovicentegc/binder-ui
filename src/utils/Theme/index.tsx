import baseStyled, { ThemedStyledInterface } from "styled-components";

export const theme = {
  importFontFamily:
    "@import url('https://fonts.googleapis.com/css?family=IBM+Plex+Sans&display=swap');",
  useFontFamily: "font-family: IBM Plex Sans, sans-serif;",
  color: {
    primaryBlack: "#262626",
    black1: "#202020",
    black2: "#2B2B2B",
    white1: "#F7F7F7",
    white2: "#DCDDE0", // TODO: move to grey
    white3: "#F5F5F5",
    gray1: "#E9E9E9",
    gray2: "#C4C4C4",
    gray3: "#999999",
    gray4: "#4C4C4C",
    gray5: "#D9D9D9",
    gray6: "8C8C8C",
    gray7: "#4D4D4D",
    gray8: "#E6E6E6",
    gray9: "#5C5C5C",
    gray10: "#EFF6EA",
    red1: "#D24654",
    red2: "#BB3B46",
    red3: "#A52F38",
    red4: "#831D23",
    red5: "#831D23",
    orange1: "#EBA248",
    orange2: "#E2873C",
    orange3: "#D86C30",
    orange5: "#C43618",
    yellow1: "#F4DA4E",
    yellow2: "#EFC841",
    yellow3: "#EAB634",
    yellow4: "#DA7F0D",
    green1: "#5DC957",
    green2: "#4DAE48",
    green3: "#3E933A",
    green4: "#276B24",
    green5: "#0F420E",
    blue1: "#6ACEDF",
    blue2: "#59B5CF",
    blue3: "#479DBF",
    blue4: "#3584AF",
    blue5: "#12538F",
    purple1: "#5A4D80",
    rainbowColors: [
      "#CC0000",
      "#F06D00",
      "#E9CD00",
      "#00BE3A",
      "#439AE5",
      "#5400CE",
      "#5E0096",
      "#EF0067"
    ]
  }
};

export type Theme = typeof theme;
export const styled = baseStyled as ThemedStyledInterface<Theme>;
