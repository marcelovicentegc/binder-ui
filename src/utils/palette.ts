export const palette = [
  // Gray scale
  "#E9E9E9",
  "#C4C4C4",
  "#999999",
  "#4C4C4C",
  "#262626",
  // Red scale
  "#D24654",
  "#BB3B46",
  "#A52F38",
  "#831D23",
  // Orange scale
  "#EBA248",
  "#E2873C",
  "#D86C30",
  "#C9441E",
  "#C43618",
  // Yellow scale
  "#F4DA4E",
  "#EFC841",
  "#EAB634",
  "#DA7F0D",
  // Green scale
  "#5DC957",
  "#4DAE48",
  "#3E933A",
  "#276B24",
  "#0F420E",
  // Blue scale
  "#6ACEDF",
  "#59B5CF",
  "#479DBF",
  "#3584AF",
  "#12538F"
];
