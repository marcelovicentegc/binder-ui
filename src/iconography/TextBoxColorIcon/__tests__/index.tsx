import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
import { TextBoxColorIcon } from "..";
import { theme } from "../../../utils/Theme";

describe("<TextBoxColorIcon /> test case", () => {
  test("test id is in the document", () => {
    const { getByTestId } = render(<TextBoxColorIcon />);
    expect(getByTestId("textBoxColorIcon")).toBeInTheDocument();
  });

  test("renders colors passed via props", () => {
    const { getByTestId } = render(
      <TextBoxColorIcon color={theme.color.orange3} />
    );
    expect(getByTestId("textBoxColorIcon")).toMatchSnapshot();
  });
});
