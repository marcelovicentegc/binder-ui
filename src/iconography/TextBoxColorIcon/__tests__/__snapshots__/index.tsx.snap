// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextBoxColorIcon /> test case renders colors passed via props 1`] = `
<svg
  color="#D86C30"
  data-testid="textBoxColorIcon"
  fill="none"
  height="32"
  viewBox="0 0 32 32"
  width="32"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    clip-path="url(#clip0)"
  >
    <path
      d="M21.6336 21L19.7513 15.9828H12.1681L10.2857 21H8L14.5613 4H17.4387L24 21H21.6336ZM16.0134 5.89971H15.879L12.7059 14.1805H19.1866L16.0134 5.89971Z"
      fill="#262626"
    />
    <path
      clip-rule="evenodd"
      d="M28 23H4V29H28V23ZM26 25H6V27H26V25Z"
      fill="#D86C30"
      fill-rule="nonzero"
    />
    <path
      clip-rule="evenodd"
      d="M4 0V-1H6V0H4ZM4 2V11V13H6H9.08769L9.85961 11H6V2H4ZM26 2V11H22.1404L22.9123 13H26H28V11V2H26ZM28 0V-1H26V0H28Z"
      fill="#262626"
      fill-rule="evenodd"
    />
  </g>
  <defs>
    <clippath
      id="clip0"
    >
      <path
        d="M0 0H32V32H0V0Z"
        fill="white"
      />
    </clippath>
  </defs>
</svg>
`;
